Clean WasabiSetupRegtest.md (#3148)
zkSNACKs/WalletWasabi: WalletWasabi.Documentation/WasabiSetupRegtest.md

## Why do this?

RegTest is a local testing environment in which developers can almost instantly generate blocks on demand for testing events, and create private satoshis with no real-world value. Running Wasabi Backend on RegTest allows you to emulate network events and observe how the Backend and the Client react on that.
You do not need to download the blockchain for this setup!

## Setup Bitcoin Core with RegTest

> Follow [this guide](https://bitcoin.org/en/developer-examples#regtest-mode).

Author: But there is not installation in the previous link as well.

> Follow [this guide](https://bitcoin.org/en/developer-examples#regtest-mode).

Reviewer: You are right I was scrolling down and up in the doc but the tag remained in the link. 

```suggestion
Follow [this guide](https://bitcoin.org/en/developer-examples).
```

Todo:

1. Install [Bitcoin Core](https://bitcoin.org/en/bitcoin-core/) on your computer.
2. Start Bitcoin Core with: bitcoin-qt.exe -regtest then quit immediately. In this way the data directory and the config files will be generated.
```
Windows: "C:\Program Files\Bitcoin\bitcoin-qt.exe" -regtest
macOS: "/Applications/Bitcoin-Qt.app/Contents/MacOS/Bitcoin-Qt" -regtest

> Linux:

Reviewer: @MaxHillebrand do you have any idea where is this on Linux?

```

> 3. Go to Bitcoin Core settings directory. If the directory is missing run core bitcoin-qt, then quit immediately. In this way the data directory and the config files will be generated.

Reviewer 2: I got confused by what is the `settings directory`, but it is what is mentioned in lines 23-25, `/.bitcoin/`?

```
Windows: %APPDATA%\Bitcoin\
macOS: $HOME/Library/Application Support/Bitcoin/
Linux: $HOME/.bitcoin/
```
4. Edit bitcoin.conf file and add these lines there:

> ```

Reviewer 2: ```suggestion
```json
```

> regtest.rpcuser = 7c9b6473600fbc9be1120ae79f1622f42c32e5c78d

Reviewer 2: is this the default password and user, or one randomly specified by @molnard, should the user change this password?

regtest.rpcpassword = 309bc9961d01f388aed28b630ae834379296a8c8e3
```
5. Start Bitcoin Core with: bitcoin-qt.exe -regtest.

> 6. Do not worry about "Syncing Headers" just press the Hide button.

Reviewer 2: ```suggestion
6. Do not worry about "Syncing Headers" just press the Hide button. Because you run on Regtest, no Mainnet blocks will be downloaded.
```

7. Go to MainMenu / Window / Console.
8. Generate a new address with:
`getnewaddress`
9. Generate the first 101 blocks with:

> `generatetoaddress 101 replace_new_address_here`

Reviewer 2: ```suggestion
`generatetoaddress 101 <replace_new_address_here>`
```

10. Now you have your own Bitcoin blockchain and you are good to go. You can create transactions with the Send button and confirm with:

> `generatetoaddress 1 replace_new_address_here`

Reviewer 2: ```suggestion
`generatetoaddress 1 <replace_new_address_here>`
```

> 10. Now you have your own Bitcoin blockchain and you are good to go. You can create transactions with the Send button and confirm with:

Author: At first I thought it was a mistake, then I realized it was not.

> 10. Now you have your own Bitcoin blockchain and you are good to go. You can create transactions with the Send button and confirm with:

Reviewer: ```suggestion
10. Now you have your own Bitcoin blockchain and you are a God there - try to resist the insurmountable temptation to start your own shit coin. You can create transactions with the Send button and confirm with:
```

## Setup Wasabi Backend

Here you will have to build from source, follow [these instructions here](https://github.com/zkSNACKs/WalletWasabi#build-from-source-code).

Todo:

> 1. Go to WalletWasabi\WalletWasabi.Backend folder.

Reviewer 2: ```suggestion
1. Go to `WalletWasabi\WalletWasabi.Backend` folder.
```

2. Open the command line and enter:
`dotnet run`

> 3. You will get some errors but the data directory created. Stop the backend if it is still running with CTRL-C.

Reviewer 2: ```suggestion
3. You will get some errors, but the data directory will be created. Stop the backend if it is still running with CTRL-C.
```

4. Go to the Backend folder:
```
Windows: "C:\Users\{your username}\AppData\Roaming\WalletWasabi\Backend"
macOS: "/Users/{your username}/.walletwasabi/backend"
Linux: "/Home/.walletwasabi/backend"
```

> 5. Edit Config.json file by replacing everything with:

Reviewer 2: ```suggestion
5. Edit `Config.json` file by replacing everything with:
```

> Linux: "/Home/.walletwasabi/backend"

Author: I don't know, I got it from here https://docs.wasabiwallet.io/FAQ/FAQ-UseWasabi.html#where-can-i-find-the-wasabi-data-folder

> Linux: "/Home/.walletwasabi/backend"

Reviewer: ```suggestion
Linux: "/home/{your username}/.walletwasabi/backend"
```

macOS: "/Users/{your username}/.walletwasabi/backend"

> Linux: "/Home/.walletwasabi/backend"

Reviewer 3: ```suggestion
Linux: "`/home/user/.walletwasabi/backend"
```

Reviewer: ```suggestion
Linux: "/home/{your username}/.walletwasabi/backend"
```

  "RegTestBitcoinCoreRpcEndPoint": "127.0.0.1:18443"
}
```

> 6. Edit one line in CcjRoundConfig.json file. With this the Coordinator waits only 2 participants for CoinJoin.

Reviewer 2: ```suggestion
6. Edit one line in `CcjRoundConfig.json` file. With this the Coordinator waits only 2 participants for CoinJoin.
```

```
"AnonymitySet": 2,
```
7. Start Bitcoin Core in RegTest.
8. Go to WalletWasabi\WalletWasabi.Backend folder.
9. Open the command line and enter:
`dotnet build`

> `dotnet run --no-build`

Reviewer 2: Why not do only `dotnet run`?

Reviewer: ACK.

10. Now the Backend is generating the filters and it is running. (You can quit with CTRL-C any time)

## Setup Wasabi Client

Todo:

> 1. Go to WalletWasabi\WalletWasabi.Gui folder.

Reviewer 2: ```suggestion
1. Go to `WalletWasabi\WalletWasabi.Gui` folder.
```

2. Open the command line and run the Wasabi Client with:
`dotnet build`

> `dotnet run --no-build`

Reviewer 2: Why not only `dotnet run`?

Reviewer: ACK.
